Custom service name (#15396)
Homebrew/brew: docs/Formula-Cookbook.md

#### Service block methods

> This table lists the options you can set within a `service` block. The `run` or `name` field must be defined inside the service block. The `run` field indicates what command to run and is required before using fields other than `name`.

Reviewer: ```suggestion
This table lists the options you can set within a `service` block. The `run` or `name` field must be defined inside the service block. The `run` field indicates what command to run and is required unless a `name` of an existing service is defined.
```

Reviewer: The way I read the original the following are all valid:
```rb
service do
  name "test"
  run "a"
  something_else true
end
```
```rb
service do
  name "test"
  something_else true
end
```
```rb
service do
  run "a"
  something_else true
end
```

Author: That's right but the following is not valid.

```rb
service do
  name "test"
  something_else true
end
```

| `process_type`          | -            |  yes  | no-op | type of process to manage: `:background`, `:standard`, `:interactive` or `:adaptive`
| `macos_legacy_timers`   | -            |  yes  | no-op | timers created by `launchd` jobs are coalesced unless this is set
| `sockets`               | -            |  yes  | no-op | socket that is created as an accesspoint to the service
| `plist_name`            | `"homebrew.mxcl.#{formula.name}"` | yes | no | name of the `launchd` job

> | `service_name`          | `"homebrew.#{formula.name}"` | no | yes | name of the `systemd` job

Author: I wonder if the markdown linter will choke on this.
